tracer/src/Datadog.Tracer.Native/method_rewriter.cpp

```diff
@@ -701,6 +702,76 @@ HRESULT TracerMethodRewriter::Rewrite(RejitHandlerModule* moduleHandler, RejitHa
     return S_OK;
 }
 
+std::tuple<WSTRING, WSTRING>
+TracerMethodRewriter::GetResourceNameAndOperationName(const ComPtr<IMetaDataImport2>& metadataImport,
+                                                      const FunctionInfo* caller, TracerTokens* tracerTokens) const
+{
+    const BYTE* data = nullptr;
+    ULONG pcbData = 0;
+    auto hr = metadataImport->GetCustomAttributeByName(caller->id, tracerTokens->GetTraceAttributeType().data(),
+                                                       reinterpret_cast<const void**>(&data), &pcbData);
+    WSTRING resourceName;
+    WSTRING operationName;
+
+    // Parse the TraceAttribute
+    if (hr == S_OK)
+    {
+        PCCOR_SIGNATURE signature = data;
+        signature += 2; // skip prolog
+        const ULONG numOfNamedArgs{CorSigUncompressData(signature)};
+        signature += 1; // skip fixed arguments length
+
+        for (ULONG argIndex = 0; argIndex < numOfNamedArgs; argIndex++)
+        {
+            signature += 2; // skip FIELD/PROPERTY and ELEM
+
+            ULONG argNameLength{CorSigUncompressData(signature)}; // length of the name
+
+            // OperationName (13 characters), ResourceName (12 characters)
+            if (argNameLength != 12 && argNameLength != 13)
+            {
+                Logger::Error("TracerMethodRewriter::Rewrite: Failed to parse Trace Attribute for ",
+                              " token=", caller->id, " caller_name=", caller->type.name, ".", caller->name, "()");
+                break;
+            }
+
+            const bool isOperationName = argNameLength == 13;
+            signature += argNameLength; // skip the argument name
+            const auto value = GetStringValueFromBlob(signature);
+            if (isOperationName)
+            {
+                operationName = value;
+            }
+            else
+            {
+                resourceName = value;
+            }
+        }
+    }
+
+    if (resourceName.empty())
+    {
+        if (caller->type.name.empty())
+        {
+            resourceName = caller->name;
+        }
+        else
+        {
+            resourceName =
+                caller->type.name.empty()
```

Comment:
nit: I think you could remove the if/else branches and just use the ternary operator